### PR TITLE
Convenient Session Retrieval

### PR DESCRIPTION
**Pull request by gsteel**
|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | yes
| QA            | no

### Description

Adds simple, static methods to a new utility class to retrieve the session from a request so that users don't have to assert the return value of `getAttribute`.

This is purely convenience for consumers…

### CODE REVIEW

File: docs/book/middleware.md [hunk 1]
```diff
@@ -86,3 +86,37 @@ $app->post('/login', [
...
+    $session = SessionRetrieval::fromRequest($request);
+} catch (SessionNotInitializedException $error) {
+    // ... Handle missing session
+}
+
+// Optional retrieval without an exception
+
+$session = SessionRetrieval::fromRequestOrNull($request);
+
+```
```
Comment (weierophinney) [inline docs/book/middleware.md]: I'd recommend making this an example of a `RequestHandlerInterface` implementation; that will make the example more familiar and accessible to readers.

Further, I'd show using `fromRequestOrNull()` and/or `try { fromRequest() } catch (SessionNotInitializedException) { }` to redirect to a login page or similar, as the typical issue with a missing session is a lack of authenticated identity.

File: test/SessionRetrievalTest.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+namespace MezzioTest\Session;
+
+use Laminas\Diactoros\ServerRequest;
+use Mezzio\Session\Exception\SessionNotInitializedException;
+use Mezzio\Session\SessionInterface;
+use Mezzio\Session\SessionMiddleware;
+use Mezzio\Session\SessionRetrieval;
+use PHPUnit\Framework\TestCase;
+
+final class SessionRetrievalTest extends TestCase
```
Comment (Ocramius) [inline test/SessionRetrievalTest.php]: `/** @covers ... */` would be nice

File: src/SessionRetrieval.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mezzio\Session;
+
+use Mezzio\Session\Exception\SessionNotInitializedException;
+use Psr\Http\Message\ServerRequestInterface;
+
+final class SessionRetrieval
```
Comment (Ocramius) [inline src/SessionRetrieval.php]: `RetrieveSession`, rather than `SessionRetrieval`? That makes it a bit nicer when looking at the call-site

**Review by Ocramius — APPROVED** [review]

File: src/SessionRetrieval.php (hunk 3 above)
Comment (gsteel) [reply src/SessionRetrieval.php]: Yes! Much nicer 👍

**Comment by gsteel** [comment]
Good to merge and release?

**Review by Ocramius — APPROVED** [review]
:ship:

**Pull request merged**